### PR TITLE
Make `wasmi_execution` public to use it from tests

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by jimpo** [comment]
Can you give an example of usage? I'd prefer not to do this and instead expose a method that accepts a `WasmExecutionMethod`.

**Comment by bkchr** [comment]
What kind of example usage? I have a wasm blob and I want to execute a function in this blob.

File: core/executor/src/wasmi_execution.rs [hunk 1]
```diff
@@ -683,10 +683,24 @@ mod tests {
 		let mut ext = TestExternalities::default();
 		let test_code = WASM_BINARY;
 
-		let output = call(&mut ext, 8, &test_code[..], "test_empty_return", &[]).unwrap();
+		let output = call(&mut ext, 8, &test_code[..], "test_empty_return", &[0]).unwrap();
```
Comment (jimpo) [inline core/executor/src/wasmi_execution.rs]: Is there a reason for the argument?

File: core/executor/src/lib.rs [hunk 2]
```diff
@@ -50,6 +50,32 @@ pub use primitives::traits::Externalities;
...
+	call_data: &[u8],
+	execution_method: WasmExecutionMethod,
+	ext: &mut E,
+	code: &[u8],
+	heap_pages: u64,
+) -> error::Result<Vec<u8>> {
+	use wasm_runtime::WasmRuntime;
+
+	// make sure this is updated, when there are more execution methods
+	let WasmExecutionMethod::Interpreted = execution_method;
```
[the remark this answers is not in the document]
Comment (bkchr) [reply core/executor/src/lib.rs]: That functions loads the blob from the storage...

**Pull request merged**